src/odemis/gui/cont/tabs.py

```diff
@@ -666,27 +665,83 @@ def Show(self, show=True):
-        if main_data.role in ("cryo-secom",):
-            return 2
-        else:
-            return None
+    def _show_acquired_stream(self, acquired_stream, selected_view):
+        """
+        Create a stream controller for the given stream and view and add it to the stream bar controller list of stream controllers
+        :param acquired_stream: the static stream to show
+        :param selected_view: the view to show it on
+        """
+        stream_cont = StreamController(self.panel.pnl_cryosecom_acquired, acquired_stream, self.tab_data_model,
+                                       show_panel=True, view=selected_view, sb_ctrl=self._acquired_stream_controller)
+        selected_view.addStream(acquired_stream)
+
+        stream_cont.stream_panel.set_visible(True)
+        stream_cont.stream_panel.show_remove_btn(True)
+        self._acquired_stream_controller.stream_controllers.append(stream_cont)
+
+    def _on_current_feature_changes(self, feature):
+        """
+        Handle switching the acquired streams appropriate to the current feature
+        :param feature: (CryoFeature or None) the newly selected current feature
+        """
+        self._acquired_stream_controller.clear(clear_model=False)
+        # show the feature streams on the acquired view
+        view = self.tab_data_model.views.value[1]
+        acquired_streams = feature.streams.value if feature else []
+        for stream in acquired_streams:
+            self._show_acquired_stream(stream, view)
+
+        # show the overview maps (if any) on the overview view
+        view = self.tab_data_model.views.value[0]
+        for stream in self.tab_data_model.overviewStreams.value:
+            self._show_acquired_stream(stream, view)
+
+    def _on_acquired_streams(self, streams):
+        """
+        Filter out deleted acquired streams (features and overview) from their respective origin
+        :param streams: list(Stream) updated list of tab streams
+        """
+        # Get all acquired streams from features list and overview streams
+        acquired_streams = set()
+        for feature in self.tab_data_model.main.features.value:
+            [acquired_streams.add(s) for s in feature.streams.value]
+        [acquired_streams.add(s) for s in self.tab_data_model.overviewStreams.value]
+
+        filter_streams = acquired_streams.difference(set(streams))
+
+        for st in filter_streams:
+            if st in self.tab_data_model.overviewStreams.value:
+                self.tab_data_model.overviewStreams.value.remove(st)
+                # Remove from chamber tab too
+                chamber_tab = self.main_data.getTabByName("cryosecom_chamber")
+                chamber_tab.remove_overview_stream(st)
+            else:
+                feature = next((f for f in self.tab_data_model.main.features.value if st in f.streams.value), None)
+                if feature:
+                    feature.streams.value.remove(st)
```

Comment:
You actually want to remove the stream from *all* the features... although, normally it's only present in one of them.
You can write simpler code if you think of "remove the stream from all the features":
```
for feature in self.tab_data_model.main.features.value:
    if st in feature.streams.value:
        feature.streams.value.remove(st)
```